consent/api/permissions.py

```diff
@@ -14,10 +14,16 @@ class IsUserInRequest(permissions.BasePermission):
     """
 
     def has_permission(self, request, view):
-        if request.method == 'GET':
-            username_in_request = request.query_params.get('username')
+        if request.method in ['GET', 'DELETE']:
+            username_in_request = request.query_params.get(
+                'username',
+                request.data.get('username', '')
+            )
         else:
-            username_in_request = request.data.get('username')
+            username_in_request = request.data.get(
+                'username',
+                request.query_params.get('username', '')
+            )
```

Comment:
This is more about being cautious and imitating the way we extract certain parameters in [`get_required_query_params`](https://github.com/edx/edx-enterprise/blob/master/consent/api/v1/views.py#L95). It may be useful to write a function to reuse this type of extraction, so... *note to self: write a function for this*.

Reply:
Sounds good! That function should probably use [`QUERY_PARAM_METHODS`](https://github.com/edx/edx-enterprise/blob/master/consent/api/v1/views.py#L81) (which would need to be available globally/outside the definition of `DataSharingConsentView`)  instead of explicitly listing `GET` and `DELETE`.

Reply:
Actually not sure if the function itself should be using that, because what I had in mind of abstracting with this function was the actual body of those `if`/`else` statements. I'm not sure if we'd use that function here if it was extracting things like `program_uuid` and stuff. So instead I'm simply thinking of reusing this:

```python
def get_request_value(request, value, default=None):
    return request.query_params.get(
        value,
        request.data.get(value, default)
    )
```

Reply:
Ok nevermind, yeah we can use `QUERY_PARAM_METHODS`, sorry :P

Reply:
@UmanShahzad If `QUERY_PARAM_METHODS` is defined as `{'GET', 'DELETE'}` (and is available globally/outside the definition of `DataSharingConsentView`), we could do the following, right?

```python
from ... import QUERY_PARAM_METHODS

# ...

def get_request_value(request, key, default=None):
    """
    Get the value in the request, either through query parameters or posted data, from a key.

    :param request: The request from which the value should be gotten from.
    :param key: The key to use to get the desired value.
    :param default: The backup value to use in case the input key cannot help us get the value.
    :return: The value we're looking for.
    """
    if request.method in QUERY_PARAM_METHODS:
        return request.query_params.get(key, request.data.get(key, default))
    else:
        return request.data.get(key, request.query_params.get(key, default))
```

That's what I was getting at :) Sorry if that wasn't clear!

Reply:
Ah I just missed this comment while addressing the review.

Yeah we can do this, but I'm not sure what would be a reasonable place for `QUERY_PARAM_METHODS` to live, that's all. Maybe in `enterprise.constants`? Either way this is not that big of a deal because it's not being reused right now. If you're okay with it, we can come back to this later once it becomes necessary to reuse.

Reply:
Fine with me :)